src/nvim/screen.c

```diff
@@ -6978,12 +6978,14 @@ void showruler(int always)
   } else
     win_redr_ruler(curwin, always);
 
-  if (need_maketitle
-      || (p_icon && (stl_syntax & STL_IN_ICON))
-      || (p_title && (stl_syntax & STL_IN_TITLE))
-      )
+  if (
+    (p_icon && (stl_syntax & STL_IN_ICON)) || 
```

Comment:
`||` must be at the start, not at the end: http://neovim.io/develop/style-guide.xml?showone=Boolean_Expressions#Boolean_Expressions.